lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1236 @@
+        )
+        return True
+    return False
+
+
+def update_instance_state(client, module, instance, state):
+    changed = False
+    if state in ['rebooted', 'restarted']:
+        changed |= reboot_running_db_instance(client, module, instance)
+    if state in ['started', 'running', 'stopped']:
+        changed |= start_or_stop_instance(client, module, instance, state)
+    return changed
+
+
+def reboot_running_db_instance(client, module, instance):
+    parameters = {'DBInstanceIdentifier': instance['DBInstanceIdentifier']}
+    if instance['DBInstanceStatus'] in ['stopped', 'stopping']:
+        call_method(client, module, 'start_db_instance', parameters)
+    if module.params.get('force_failover') is not None:
+        parameters['ForceFailover'] = module.params['force_failover']
+    results, changed = call_method(client, module, 'reboot_db_instance', parameters)
+    return changed
+
+
+def start_or_stop_instance(client, module, instance, state):
+    changed = False
+    parameters = {'DBInstanceIdentifier': instance['DBInstanceIdentifier']}
+    if state == 'stopped' and instance['DBInstanceStatus'] not in ['stopping', 'stopped']:
+        if module.params['db_snapshot_identifier']:
+            parameters['DBSnapshotIdentifier'] = module.params['db_snapshot_identifier']
+        result, changed = call_method(client, module, 'stop_db_instance', parameters)
+    elif state == 'started' and instance['DBInstanceStatus'] not in ['available', 'starting', 'restarting']:
+        result, changed = call_method(client, module, 'start_db_instance', parameters)
+    return changed
+
+
+def main():
+    arg_spec = dict(
+        state=dict(choices=['present', 'absent', 'running', 'started', 'stopped', 'rebooted', 'restarted'], default='present'),
+        creation_source=dict(choices=['snapshot', 's3', 'instance']),
+        force_update_password=dict(type='bool', default=False),
+        promote=dict(type='bool', default=False),
+        purge_cloudwatch_logs_exports=dict(type='bool', default=True),
+        purge_tags=dict(type='bool', default=True),
+        wait=dict(type='bool', default=True),
+    )
+
+    parameter_options = dict(
+        allocated_storage=dict(type='int'),
+        allow_major_version_upgrade=dict(type='bool'),
+        apply_immediately=dict(type='bool', default=False),
+        auto_minor_version_upgrade=dict(type='bool'),
+        availability_zones=dict(type='list', aliases=['zones', 'az']),
+        backup_retention_period=dict(type='int'),
+        ca_certificate_identifier=dict(),
+        character_set_name=dict(),
+        copy_tags_to_snapshot=dict(type='bool'),
+        db_cluster_identifier=dict(aliases=['cluster_id']),
+        db_instance_class=dict(aliases=['class']),
+        db_instance_identifier=dict(required=True, aliases=['instance_id', 'id']),
+        db_name=dict(),
+        db_parameter_group_name=dict(),
+        db_security_groups=dict(),
```

Comment:
This needs to be type='list' -- the doc mentions it's a list but it's not forced as such which can yield the error ``Invalid type for parameter DBSecurityGroups, value: sg-1234567890, type: <type 'str'>, valid types: <type 'list'>, <type 'tuple'>``